packages/block-editor/src/utils/test/get-font-styles-and-weights.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { getFontStylesAndWeights } from '../get-font-styles-and-weights';
+
+describe( 'getFontStylesAndWeights', () => {
+	it( 'should return empty object if fontFamilyFaces is not available', () => {
+		expect( getFontStylesAndWeights() ).toEqual( {} );
+	} );
+
+	it( 'should return correct font styles and weights based on available options', () => {
+		const fontFamilyFaces = [
+			{
+				fontFamily: 'ABeeZee',
+				fontStyle: 'italic',
+				fontWeight: '400',
+				src: 'http://www.wordpress.org/wp-content/uploads/fonts/esDT31xSG-6AGleN2tCkkJUCGpG-GQ.woff2',
+			},
+		];
+		expect( getFontStylesAndWeights( fontFamilyFaces ) ).toEqual( {
+			fontStyles: [
+				{
+					name: 'Italic',
+					value: 'italic',
+				},
+			],
+			fontWeights: [
+				{
+					name: 'Regular',
+					value: '400',
+				},
+			],
+		} );
+	} );
+
+	it( 'should return all available style and weight options if font weights string includes a space', () => {
+		const fontFamilyFaces = [
+			{
+				fontFamily: 'Inter',
+				fontStyle: 'normal',
+				fontWeight: '100 900',
+				src: 'http://www.wordpress.org/wp-content/uploads/fonts/Inter-VariableFont_slntwght.ttf',
+			},
+		];
+
+		expect( getFontStylesAndWeights( fontFamilyFaces ) ).toEqual( {
+			fontStyles: [
+				{
+					name: 'Regular',
+					value: 'normal',
+				},
+				{
+					name: 'Italic',
+					value: 'italic',
+				},
+			],
+			fontWeights: [
+				{
+					name: 'Thin',
+					value: '100',
+				},
+				{
+					name: 'Extra Light',
+					value: '200',
+				},
+				{
+					name: 'Light',
+					value: '300',
+				},
+				{
+					name: 'Regular',
+					value: '400',
+				},
+				{
+					name: 'Medium',
+					value: '500',
+				},
+				{
+					name: 'Semi Bold',
+					value: '600',
+				},
+				{
+					name: 'Bold',
+					value: '700',
+				},
+				{
+					name: 'Extra Bold',
+					value: '800',
+				},
+				{
+					name: 'Black',
+					value: '900',
+				},
+			],
+		} );
+	} );
```

Comment:
Wrote a test for [this](https://github.com/WordPress/gutenberg/pull/61915#pullrequestreview-2094185074) — it does fail.

> Install 'Piazzolla' font with normal-100, nomal-200 and italic-800 and italic-900 styles.
> Expected:
> The font styles available should be added only if there are font faces available for those styles (variable or not) or if it's a system font.

```suggestion
	} );

	it( 'should return only provided font styles and weights', () => {
		const fontFamilyFaces = [
			{
				fontFamily: 'Piazzolla',
				fontStyle: 'normal',
				fontWeight: '400',
				src: 'http://www.wordpress.org/wp-content/uploads/fonts/N0b52SlTPu5rIkWIZjVKKtYtfxYqZ4RJBFzFfYUjkSDdlqZgy7LYxnL31AHfAAy5.woff2',
			},
			{
				fontFamily: 'Piazzolla',
				fontStyle: 'normal',
				fontWeight: '900',
				src: 'http://www.wordpress.org/wp-content/uploads/fonts/N0b52SlTPu5rIkWIZjVKKtYtfxYqZ4RJBFzFfYUjkSDdlqZgy7JxwXL31AHfAAy5.woff2',
			},
			{
				fontFamily: 'Piazzolla',
				fontStyle: 'italic',
				fontWeight: '300',
				src: 'http://www.wordpress.org/wp-content/uploads/fonts/N0b72SlTPu5rIkWIZjVgI-TckS03oGpPETyEJ88Rbvi0_TzOzKcQhcSx3gD9BRy5m5M.woff2',
			},
			{
				fontFamily: 'Piazzolla',
				fontStyle: 'italic',
				fontWeight: '900',
				src: 'http://www.wordpress.org/wp-content/uploads/fonts/N0b72SlTPu5rIkWIZjVgI-TckS03oGpPETyEJ88Rbvi0_TzOzKcQhTO23gD9BRy5m5M.woff2',
			},
		];

		expect( getFontStylesAndWeights( fontFamilyFaces ) ).toEqual( {
			fontStyles: [
				{
					name: 'Regular',
					value: 'normal',
				},
				{
					name: 'Italic',
					value: 'italic',
				},
			],
			fontWeights: [
				{
					name: 'Light',
					value: '300',
				},
				{
					name: 'Black',
					value: '900',
				},
			],
		} );
	} );
```

Reply:
Thanks for this. I've switched the less complicated `ABeeZee` test with your `Piazzolla` test above in https://github.com/WordPress/gutenberg/pull/61915/commits/1ca996bfd54ada3246596aab3ae22586652227b3.